src/components/Calender/DateDisplay.tsx

```diff
@@ -20,7 +21,11 @@ export const DateDisplay = (props: DateDisplayProps) => {
     const { firstDay, tasks } = props;
     const initialDate = new Date(year, month, 1);
     const currentDate = useSelector((state) => state.calendarReducer.today);
-    const selectedDate = useSelector((state) => state.calendarReducer.selectedDate?.date);
+
+    const selectedDate =
+        props.reducerType === 'calendarReducer'
```

Comment:
instead of calendarReducer, I'd say you can differentiate by the reducer you need to use by saying it's home page or if it's swap page. thus instead of using reducerType, it would really be type which is generally more typical